### PR TITLE
D 168

### PR DESCRIPTION
**Pull request by RotateAt60MPH**
Refactored to use new DHL Apigee API - authentication and Duty/Tax Calculator.

### CODE REVIEW

File: Helper/Data.php [hunk 1]
```diff
@@ -19,8 +19,13 @@ class Data extends AbstractHelper
     const STASH_SANDBOX_URL = 'https://stash-sandbox.gointerpay.net/';
     const SANDBOX_API_URL = 'https://checkout-sandbox.gointerpay.net/v2.19/';
 
-    const DHL_API_URL = 'https://api.dhlecommerce.com/';
-    const DHL_SANDBOX_API_URL = 'https://api-sandbox.dhlecommerce.com/';
+    //const DHL_API_URL = 'https://api.dhlecommerce.com/';
+    //const DHL_SANDBOX_API_URL = 'https://api-sandbox.dhlecommerce.com/';
```
Comment (brenleydueck) [inline Helper/Data.php]: Can take out these comments

File: Model/DutyCalculator.php [hunk 2]
```diff
@@ -144,12 +159,12 @@ public function handleTaxApplicability($address, $apply)
...
-               "(shipment address)");
-           $quote->setReachDuty($this->checkoutSession->getReachDuty());
-           $quote->setBaseReachDuty($this->checkoutSession->getBaseReachDuty());
-           $quote->setDhlQuoteId($this->checkoutSession->getDhlQuoteId());
+            $this->_logger->debug("Duty should be applied (due to user selection or the value of the 'apply' ".
+                " parameter that is passed to this function or because duty is not optional for the chosen country ".
+                "(shipment address)");
+            $quote->setReachDuty($this->checkoutSession->getReachDuty());
+            $quote->setBaseReachDuty($this->checkoutSession->getBaseReachDuty());
+            $quote->setDhlQuoteId($this->checkoutSession->getDhlQuoteId());
```
Comment (brenleydueck) [inline Model/DutyCalculator.php:165]: looks like just formatting changes

File: Test/Unit/Model/DutyCalculatorTest.php [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+            $this->scopeConfig,
+            $this->storeManager,
+            $this->priceCurrency,
+            $this->csvHsCodeFactory,
+            $this->regionModel,
+            $this->httpRestFactory,
+            $this->response,
+            $this->logger);
+
+        $json = '{
```
Comment (brenleydueck) [inline Test/Unit/Model/DutyCalculatorTest.php:134]: could put this in seperate file to shorten up the test

File: Model/DutyCalculator.php (hunk 2 above)
Comment (RotateAt60MPH) [reply Model/DutyCalculator.php:165]: I tried putting in explicit spaces, but on refresh, it still shows old formatting.

File: Test/Unit/Model/DutyCalculatorTest.php (hunk 3 above)
Comment (RotateAt60MPH) [reply Test/Unit/Model/DutyCalculatorTest.php:134]: Will be resolved in this Sprint.

**Review by brenleydueck — APPROVED** [review]

**Review by brenleydueck — COMMENTED** [review]
Looks good

File: Helper/Data.php [hunk 4]
```diff
@@ -462,7 +466,12 @@ public function getDhlApiUrl()
      */
     public function getDhlApiKey()
     {
-        return $this->getConfigValue(self::DHL_API_KEY, $this->storeManager->getStore()->getId());
+        if ($this->getConfigValue(self::CONFIG_API_MODE, $this->storeManager->getStore()->getId())
```
Comment (brenleydueck) [inline Helper/Data.php]: its a little weird to use the getId in this way.. could the id be 1 and not be in sandbox mode?

**Review by brenleydueck — APPROVED** [review]

**Review by brenleydueck — APPROVED** [review]
Looks good

**Comment by brenleydueck** [comment]
Looks good

**Pull request merged**